### DISCUSSION TITLE
Available preview targets should dynamically update when the entry type (or other content) is changed

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/cms#10488] [Ideas] [answered]
### Description

Craft Pro allows you to define multiple preview targets with custom URL formats. One feature that (as far as I know) is undocumented is that if a preview target template results in an empty string, it will not be shown in the frontend. We're using this to dynamically display different preview targets based on the entry type. For example, here is a preview target we have defined for a `news` section:

```twig
{% if object.type.handle != 'short_news' %}{{ object.url }}{% endif %}
```

This preview target will only be displayed if the current entry type is not `short_news`. The problem is that changing the entry type does not trigger an update of the preview target links/buttons. So if a `news` entry is switched from a different entry type to `short_news`, this preview target will still be available until the entry is saved and thereby reloaded. Since we throw a 404 error for `short_news`, this can result in a confusing editor experience.

This is admittedly a very low-priority issue, but it would be nice if preview targets could be dynamically updated when the entry type changes. Of course, theoretically _any_ attribute or field change could result in the preview targets changing, so maybe they need to be updated after every change? In any case, having preview targets differ based on entry type seems to be the most common scenario.

### Additional info

- Craft version: 3.7.30.1
- PHP version: 8.1
- Database driver & version: MySQL 8.0

**Comment by brandonkelly** [comment, marked as answer]
We only filter out empty targets to prevent Live Preview from showing up for entries that don’t actually have URLs. We did not mean for the feature to be exploited to create conditional preview targets. I’m not against us adding the feature, but considering this a feature request rather than a bug because it’s currently working as expected for the intended use case.

**Reply by MoritzLost** [reply]
@brandonkelly Fair enough, thanks for the reply! Would be nice to get this feature at some point.
